sigoptlite/sigoptlitetest/endpoints/test_best_assignments.py

```diff
@@ -127,108 +163,146 @@ def test_multimetric_thresholds(self, threshold_y1, threshold_y2):
         dict(name="y2", objective="minimize", threshold=threshold_y2),
       ],
       observation_budget=123,
+      type="random",
     )
-    experiment = LocalExperimentBuilder(experiment_meta)
-    pareto_optimal_observations = self.create_observations_with_values_in_simplex(experiment, on_boundary=True)
-    non_optimal_observations = self.create_observations_with_values_in_simplex(experiment, on_boundary=False)
-    observations = pareto_optimal_observations + non_optimal_observations
-    random.shuffle(observations)
+    e = self.conn.experiments().create(**experiment_meta)
+
+    num_optimal = 12
+    pareto_optimal_assignments = self.generate_random_assignments(experiment_meta, num=num_optimal)
+    non_optimal_assignments = self.generate_random_assignments(experiment_meta, num=num_optimal)
+    assignments_list = pareto_optimal_assignments + non_optimal_assignments
+
+    pareto_optimal_values = self.create_values_in_simplex(on_boundary=True, num=num_optimal)
+    non_optimal_values = self.create_values_in_simplex(on_boundary=False)
+    values_list = pareto_optimal_values + non_optimal_values
+
+    for assignments, values in sorted(zip(assignments_list, values_list), key=lambda _: random.random()):
+      self.conn.experiments(e.id).observations().create(
+        assignments=assignments,
+        values=values,
+      )
+
+    best_assignments = self.conn.experiments(e.id).best_assignments().fetch()
+    best_assignments_list = list(best_assignments.iterate_pages())
+    self.assert_best_assignments_are_sorted(e, best_assignments_list)
 
     if threshold_y1 is None:
       threshold_y1 = 0
-
     if threshold_y2 is None:
       threshold_y2 = 0
-
-    best_assignments_logger = BestAssignmentsLogger(experiment)
-    best_observations_from_logger = best_assignments_logger.fetch(observations)
-    self.assert_observations_are_sorted(best_observations_from_logger, experiment)
-
-    expected_pareto_optimal_observations = [o.get_client_observation(experiment) for o in pareto_optimal_observations]
-    for observation in expected_pareto_optimal_observations:
-      if observation["values"][0]["value"] >= threshold_y1 and observation["values"][1]["value"] <= threshold_y2:
-        assert observation in best_observations_from_logger
+    for best_assignment in best_assignments_list:
+      best_values = [dict(name=value.name, value=value.value) for value in best_assignment.values]
+      if best_values[0]["value"] >= threshold_y1 and best_values[1]["value"] <= threshold_y2:
+        assert best_values in pareto_optimal_values
       else:
-        assert observation not in best_observations_from_logger
+        assert best_values not in pareto_optimal_values
 
   def test_metric_constraints(self):
```

Comment:
For metric constraints, we should also test the trivial case of non of the observations satisfying the constraint, where we return an empty list.